src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddBaseCurrencySubscriber.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Form\EventSubscriber;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AddBaseCurrencySubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $resource = $event->getData();
+        $disabled = false;
+
+        if ($resource instanceof ChannelInterface) {
+            $disabled = null !== $resource->getId();
+        } elseif (null !== $resource) {
+            throw new UnexpectedTypeException($resource, ChannelInterface::class);
+        }
```

Comment:
What would you say for:

```php
private function getDisabledOption($resource)
{
    if ($resource instanceof ChannelInterface) {
        return null !== $resource->getId();
    }

    if (null === $resource) {
        return false;
    }

    throw new UnexpectedTypeException($resource, ChannelInterface::class);
}
```
? We could avoid this nasty `elseif` and make code more readable.